### ISSUE TITLE
Improve public IP resolver logging

### ISSUE THREAD

**Issue by skitt** [submariner-io/submariner#2350]
Currently, when public IP resolvers cause errors, we log those errors, but we don’t log the overall state — so just looking at the logs is misleading.

We should either log success at the same severity, or only log errors if we can’t get a public IP address _at all_.

**Comment by nyechiel** [comment]
@sridhargaddam FYI

**Comment by sridhargaddam** [comment]
> We should either log success at the same severity, 

With the recent enhancements to the public-ip watcher where we use random resolver each time and with [10 external servers](https://github.com/submariner-io/submariner/blob/e3d9a03dcdec50fd60d9ce971888ca68764632fc/pkg/endpoint/public_ip.go#L58) the likelyhood of failure is drastically reduced. So, if we log even success scenarios at debug level, I'm afraid the submariner-gateway log might be filled with these entries when the pod runs for couple of days.

>or only log errors if we can’t get a public IP address _at all_.

We already do this at two places.
1. When the Submariner Gateway pod comes up, it tries to resolve the public IP of the Gateway node and if its unable to resolve it, its treated as a [FATAL error](https://github.com/submariner-io/submariner/blob/e3d9a03dcdec50fd60d9ce971888ca68764632fc/main.go#L149) (see [here](https://github.com/submariner-io/submariner/blob/e3d9a03dcdec50fd60d9ce971888ca68764632fc/pkg/endpoint/local_endpoint.go#L94))  
2. If the periodic resolver cannot resolve the public IP even after trying **all the resolvers**, [even this](https://github.com/submariner-io/submariner/blob/e3d9a03dcdec50fd60d9ce971888ca68764632fc/pkg/endpoint/public_ip.go#L111) is logged [here](https://github.com/submariner-io/submariner/blob/e3d9a03dcdec50fd60d9ce971888ca68764632fc/pkg/endpoint/public_ip_watcher.go#L72)

I guess you were probably looking at the logs from Submariner 0.13 image where some of these enhancements were missing. I'll take care of back porting the necessary PRs addressing the conflicts.

**Comment by skitt** [comment]
I agree that the logs would be too verbose if we logged everything. My point regarding only logging errors if we can’t get a public IP address at all isn’t that we don’t already log overall failures, but that we should only log individual resolver errors if we fail altogether.

What I want to avoid is scenarios where individual resolver errors are logged, but nothing indicates overall success, without having to document the fact that overall errors are logged (with high severity) and that resolver errors should be ignored if they are not followed by a high severity overall error. The approaches I see to avoid that are

* if we log individual errors, log an overall success message (_i.e._ in the loop, track whether we logged a failure, and if we exit the loop successfully, log success if we logged failure)
* skip logging individual errors, and accumulate them instead, and log them if we fail to get an address altogether

I think we (as developers) should care about resolvers which cause problems, but that’s not really something the end users care about; perhaps this should be turned into a metric that can then be used in telemetry.
